pfcpiface/p4rtc.go

```diff
@@ -294,9 +341,13 @@ func (c *P4rtClient) WriteReq(update *p4.Update) error {
 		ElectionId: &c.electionID,
 		Updates:    []*p4.Update{update},
 	}
+
 	_, err := c.client.Write(context.Background(), req)
+	if err != nil {
+		return convertError(err)
+	}
 
-	return err
+	return nil
```

Comment:
If we let `convertError` handle the `err == nil` case, it would save us if the check on every call site.

```go
return convertError(err)
```

Or is that no idiomatic go error handling?

Reply:
hmm `convertError` assumes non-nil error. At least, I'm not sure how `FromError` behaves if it gets `nil`.

Reply:
So we could make the function more robust by doing the check at the start of it?

Reply:
done